ssl/packet.c

```diff
@@ -211,8 +216,28 @@ static int wpacket_intern_close(WPACKET *pkt)
                               sub->lenbytes))
             return 0;
 
-    pkt->subs = sub->parent;
-    OPENSSL_free(sub);
+    if (doclose) {
+        pkt->subs = sub->parent;
+        OPENSSL_free(sub);
+    }
+
+    return 1;
+}
+
+int WPACKET_fill_lengths(WPACKET *pkt)
+{
+    WPACKET_SUB *sub;
+
+    assert(pkt->subs != NULL);
```

Comment:
I thought the consensus was heading towards using soft asserts followed by an "if" for the same condition. That should mean that a debug build that fails the assert will abort, but a non debug build will just return from the function with an error. That at least was the approach we took in the original WPACKET implementation (as you can see elsewhere in the file).